drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -218,6 +241,12 @@ static int stm32_clock_control_init(struct device *dev)
 
 #ifdef CONFIG_CLOCK_STM32_SYSCLK_SRC_PLL
 
+#if defined(CONFIG_STM32H7_DUAL_CORE)
```

Comment:
I wonder if hsem protection is required here since only M7 run this function (and only once at boot for now)

Reply:
It is true that M4 doesn't run this function, but it is still able to access  RCC registers.
And even if today it should ot happends while M7 is configuring clocks,
I prefer to keep it in case it becomes necessary (with introduction of power management for example)